tremor-script/src/registry/custom_aggregate_fn.rs

```diff
@@ -0,0 +1,237 @@
+// Copyright 2020-2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::ast::Exprs;
+use crate::interpreter::{Cont, Env, LocalStack};
+use crate::prelude::{Builder, ExecOpts};
+use crate::registry::{FResult, FunctionError};
+use crate::{AggrType, Value};
+use beef::Cow;
+
+/// public because lalrpop
```

Comment:
I don't think all those things need to be public, lalrpop should only ever use the `Raw` versions lets make them private so they don't leak implementation and interface